### PR TITLE
[SPARK-42351][CORE] Protobuf serializer for FsHistoryProviderMetadata

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Add Protobuf serializer for `FsHistoryProviderMetadata`

### Why are the changes needed?
Support fast and compact serialization/deserialization for `FsHistoryProviderMetadata` over RocksDB.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
New UT

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala [hunk 1]
```diff
@@ -1448,6 +1449,21 @@ class KVStoreProtobufSerializerSuite extends SparkFunSuite {
     }
   }
 
+  test("FsHistoryProviderMetadata") {
+    Seq( "file:/tmp/spark-events", null).foreach { logDir =>
```
Comment (HyukjinKwon) [inline core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala:1453]:
```suggestion
    Seq("file:/tmp/spark-events", null).foreach { logDir =>
```

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto [hunk 2]
```diff
@@ -816,3 +816,9 @@ message StreamingQueryProgress {
 message StreamingQueryProgressWrapper {
   StreamingQueryProgress progress = 1;
 }
+
+message FsHistoryProviderMetadata {
+  int64 version = 1;
+  int64 ui_version = 2;
+  optional string log_dir = 3;
+}
```
Comment (techaddict) [inline core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:824]: :nit add a new line at the end

Comment (LuciferYang) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto]: done

**Comment by LuciferYang** [comment]
GA passed

**Comment by gengliangwang** [comment]
@LuciferYang So as per the comment in https://issues.apache.org/jira/browse/SPARK-41053?focusedCommentId=17651170&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17651170, let's support protobuf serliazer for all the live UI data for now. For SHS related, we can simply use json since the objects are not frequently used. The `FsHistoryProviderMetadata` is written once only.

**Comment by LuciferYang** [comment]
> @LuciferYang So as per the comment in https://issues.apache.org/jira/browse/SPARK-41053?focusedCommentId=17651170&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17651170, let's support protobuf serliazer for all the live UI data for now. For SHS related, we can simply use json since the objects are not frequently used. The `FsHistoryProviderMetadata` is written once only.

OK, close this one and https://github.com/apache/spark/pull/39944, `RocksDB.TypeAliases` also write only in Live UI scene

**Pull request closed without merging**